cpp/src/parquet/column_writer.cc

```diff
@@ -2603,7 +2622,24 @@ struct SerializeFunctor<
 template <>
 struct SerializeFunctor<::parquet::FLBAType, ::arrow::HalfFloatType> {
   Status Serialize(const ::arrow::HalfFloatArray& array, ArrowWriteContext*, FLBA* out) {
+#if ARROW_LITTLE_ENDIAN
+    return SerializeLittleEndianValues(array, array.raw_values(), out);
+#else
     const uint16_t* values = array.raw_values();
+    const int64_t length = array.length();
```

Comment:
Can we create `SerializeLittleEndianValues()` for big-endian, which calls `::arrow::bit_util::ToLittleEndian()` when `&values[i]` is called?
Then we can use it instead of storing converted values to a vector? This can reduce memory accesses.

Reply:
We already have endian-fixing logic for FLBA in the decimal serializer above. It should be factored out and reused instead of reinventing something different here.